### PR TITLE
Add ability to set repository policy

### PR DESCRIPTION
**Pull request by alexanderldavis**
This PR introduces the ability to set and update an ECR repository's `repository policy`, similar to how lifecycle policies are handled today.

Since repository policies can be set for both public and private ECR repositories, I moved the setRepositoryPolicy functionality to a `common.ts` file that is consumed by both the `ecr.ts` and `ecr_public.ts`.

I've tested this live in my own account. I was able to create a ECR repository without a repository policy, then add a repository-policy by setting the action parameter, and then edited the json and confirmed that the policy was updated by the action.

### CODE REVIEW

**Comment by kvlxj** [comment]
This feature would be very helpful. Thank you @alexanderldavis for this PR :) 

When will this be reviewed and merged ?

**Comment by alexanderldavis** [comment]
@int128 Let me know if there is something I can do to help get this merged in, or if you have concerns I can address!

File: src/common.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as core from '@actions/core'
+import {
+  ECRClient,
+  SetRepositoryPolicyCommand,
+} from '@aws-sdk/client-ecr'
```
Comment (int128) [inline src/common.ts:5]: I think it needs to define each function for ECR and ECR Public, because `SetRepositoryPolicyCommand` type may be different.

**Review by int128 — COMMENTED** [review]
Thank you for the contribution.

File: src/common.ts (hunk 1 above)
Comment (alexanderldavis) [reply src/common.ts]: @int128 Thanks! You're right, and it also made me realize that the `ECRClient` is different for public (`ECRPUBLICClient`). To make things more legible, I removed the `setRepositoryPolicy` function from common.ts and added them to the respective ecr and ecr_public files.

**Review by int128 — APPROVED** [review]

**Pull request merged**